### PR TITLE
API game resolvers

### PR DESCRIPTION
**Pull request by Nosmohta**
I've created a graphQL resolver for a `Game`. The resolver returns a deeply nested data structure that has been transformed from the API data schema into the scorenami graphQL data schema.

### CODE REVIEW

File: packages/scorenami-api/README.md [hunk 1]
```diff
@@ -33,3 +37,69 @@ query {
...
+    awayScore
+  }
+}
+```
+
+* Fragment for team details of game
+```js
+fragment gameTeamDetails on TeamGameDetails {
+  team
+  totfd
```
Comment (iansu) [inline packages/scorenami-api/README.md]: I don't like these abbreviated field names. I know they come from the API this way. We might want to change them at some point but leave it for now.

File: packages/scorenami-api/lib/utilities.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
+const axios = require('axios');
+
+const config = require('../config/config');
+
+const APICall = (resource, args) => {
```
Comment (iansu) [inline packages/scorenami-api/lib/utilities.js]: `APICall` isn't a very descriptive name. This file should probably be named something like `pro-football-api` and this method should be named `PFARequest`.

File: packages/scorenami-api/lib/utilities.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  const composedQuery = Object.assign({ api_key: process.env.PRO_FOOTBALL_API_KEY }, args);
+
+  return axios.post(`${config['apiBaseUrl']}/${resource}`, composedQuery)
+}
+
+const translateGameSchema = (dataAPI) => {
+  //console.log("API DATA: ", dataAPI.home.stats);
```
Comment (iansu) [inline packages/scorenami-api/lib/utilities.js]: Remove log.

File: packages/scorenami-api/lib/utilities.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+const config = require('../config/config');
+
+const APICall = (resource, args) => {
+  const composedQuery = Object.assign({ api_key: process.env.PRO_FOOTBALL_API_KEY }, args);
+
+  return axios.post(`${config['apiBaseUrl']}/${resource}`, composedQuery)
+}
+
+const translateGameSchema = (dataAPI) => {
```
Comment (iansu) [inline packages/scorenami-api/lib/utilities.js]: `dataAPI` should be called `responseData`. Prettier should have removed the brackets on this since there is only a single parameter. Please remove them here and elsewhere.

File: packages/scorenami-api/lib/utilities.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  //console.log("API DATA: ", dataAPI.home.stats);
+
+  const dataGraphQL = dataAPI
```
Comment (iansu) [inline packages/scorenami-api/lib/utilities.js]: `dataGraphQL` should be called whatever we're building, in this case `game`.

File: packages/scorenami-api/lib/utilities.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
+const axios = require('axios');
+
+const config = require('../config/config');
+
+const APICall = (resource, args) => {
+  const composedQuery = Object.assign({ api_key: process.env.PRO_FOOTBALL_API_KEY }, args);
+
+  return axios.post(`${config['apiBaseUrl']}/${resource}`, composedQuery)
```
Comment (iansu) [inline packages/scorenami-api/lib/utilities.js]: Missing semicolons everywhere in this file. Again Prettier should fix this. Please correct the formatting of this file and I will look into why Prettier isn't running properly.

File: packages/scorenami-api/lib/utilities.js [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+  dataGraphQL.home = translateTeamGameDetailSchema(dataAPI.home)
+  dataGraphQL.away = translateTeamGameDetailSchema(dataAPI.away)
+
+  return dataGraphQL
+}
+
+const translateTeamGameDetailSchema = (dataAPI) => {
+  const dataGraphQL = dataAPI
+  const drives = []
+  for (drive in dataAPI.drives) {
```
Comment (iansu) [inline packages/scorenami-api/lib/utilities.js]: Newlines around this `for` block and throughout this file.

File: packages/scorenami-api/lib/utilities.js [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+  const dataGraphQL = dataAPI
+  dataGraphQL.id = dataAPI.nfl_id
+
+  return dataGraphQL
+}
+
+const statsTransform = (dataAPI, idPropName) => {
+  const topKeys = Object.keys(dataAPI)
+  const propNames = topKeys.length > 0 ? Object.keys(dataAPI[topKeys[0]]) : []
+  const hashMap = propNames.map(oldPropName => {
```
Comment (iansu) [inline packages/scorenami-api/lib/utilities.js]: `oldPropName` should just be `propName` or `key`.

File: packages/scorenami-api/lib/utilities.js [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+const translatePlaySchema = (dataAPI) => {
+  const dataGraphQL = dataAPI
+  dataGraphQL.id = dataAPI.nfl_id
+
+  return dataGraphQL
+}
+
+const statsTransform = (dataAPI, idPropName) => {
```
Comment (iansu) [inline packages/scorenami-api/lib/utilities.js]: It's hard to follow what's going on in this function. Improve the naming and formatting and consider breaking it up into smaller functions.

File: packages/scorenami-api/models/game.js [hunk 10]
```diff
@@ -0,0 +1,20 @@
+const axios = require('axios');
```
Comment (iansu) [inline packages/scorenami-api/models/game.js]: Axios is no longer used in this file. It seems like maybe ESLint is not working properly either. Please make sure you've installed all the dependencies for this project.

File: packages/scorenami-api/models/game.js [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+const chalk = require('chalk');
+
+const config = require('../config/config');
+
+const { APICall, translateGameSchema } = require('../lib/utilities');
+
+const getGame = args => {
+
```
Comment (iansu) [inline packages/scorenami-api/models/game.js]: No newline.

File: packages/scorenami-api/models/schedule.js [hunk 12]
```diff
@@ -21,8 +21,8 @@ const getSchedule = args => {
       response.data.forEach(game => {
         schedule.push({
           id: game.id,
-          home: game.home,
-          away: game.away,
+          home: { team: game.home },
```
Comment (iansu) [inline packages/scorenami-api/models/schedule.js]: What does this give us? Can we dig into the team in the GraphQL query?

File: packages/scorenami-api/schema/stats-schema.js [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+  fumblesLost: Int
+}
+
+type KickingStats {
+  playNumber: String
+  name: String
+  attempts: Int
+  made: Int
+  yards: Int
+  percent: Int,
```
Comment (iansu) [inline packages/scorenami-api/schema/stats-schema.js]: Remove comma.

File: packages/scorenami-api/lib/utilities.js (hunk 2 above)
Comment (Nosmohta) [reply packages/scorenami-api/lib/utilities.js]: Changed to PFARequest.

File: packages/scorenami-api/README.md (hunk 1 above)
Comment (Nosmohta) [reply packages/scorenami-api/README.md]: Ok.

File: packages/scorenami-api/lib/utilities.js (hunk 6 above)
Comment (Nosmohta) [reply packages/scorenami-api/lib/utilities.js]: Prettier is now working properly.

File: packages/scorenami-api/lib/utilities.js (hunk 3 above)
Comment (Nosmohta) [reply packages/scorenami-api/lib/utilities.js]: Done.

File: packages/scorenami-api/lib/utilities.js [hunk 14]
```diff
@@ -0,0 +1,107 @@
+const axios = require('axios');
```
Comment (iansu) [inline packages/scorenami-api/lib/utilities.js]: You forgot to rename this file. This library is specifically only for interacting with the Pro Football API so let's rename it to `pro-football-api.js`.

File: packages/scorenami-api/README.md (hunk 1 above)
Comment (iansu) [reply packages/scorenami-api/README.md]: Maybe we should consider doing this now. It's hard to know what some of these even stand for. I'll look into that.

File: packages/scorenami-api/lib/utilities.js [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+    return {
+      [propName]: propName.replace(/_([a-z,0-9])/g, g => g[1].toUpperCase())
+    };
+  });
+  const stats = playIds.map(playId => {
+    const statData = {};
+    const statsData = statsDataAPI[playId];
+    propNameTransformMap
+      .map(propNamePair => {
+        let key = Object.keys(propNamePair)[0];
```
Comment (iansu) [inline packages/scorenami-api/lib/utilities.js]: It doesn't look like `key` ever gets reassigned. Make it a `const` instead.

File: packages/scorenami-api/lib/utilities.js [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+        const newPropName = propNamePair[key];
```
Comment (iansu) [inline packages/scorenami-api/lib/utilities.js]: This is somewhat more clear with the updated names. So you're storing a mapping of the old prop names to the new prop names. Is there a reason we can't just transform them on demand right here?

File: packages/scorenami-api/models/game.js [hunk 17]
```diff
@@ -0,0 +1,18 @@
+const chalk = require('chalk');
+
+const config = require('../config/config');
+
+const { PFARequest, translateGameSchema } = require('../lib/utilities');
+
+const getGame = args => {
+  return PFARequest('game', args)
+    .then(response => {
+      return translateGameSchema(response.data);
```
Comment (iansu) [inline packages/scorenami-api/models/game.js]: It would be really nice if this translation could happen automatically and then the request would just return the already translated response data.

File: packages/scorenami-api/README.md [hunk 18]
```diff
@@ -32,4 +32,171 @@ query {
...
+    fumblesLost
+  }
+  kicking {
+    playNumber
+    name
+    attempts
+    made
+    yards
+    percent
+    extraPointsAttempt
```
Comment (iansu) [inline packages/scorenami-api/README.md]: These should all be `extraPoint` singular, not plural.

File: packages/scorenami-api/config/schema-map.js [hunk 19]
```diff
@@ -0,0 +1,152 @@
...
+    quarter: 'quarter',
+    result: 'result',
+    penyds: 'penaltyYards',
+    ydsgained: 'yardsGained',
+    numplays: 'numberOfPlays',
+    postime: 'possessionTime',
+    plays: 'plays'
+  },
+  game: {
+    nfl_id: 'gameId',
```
Comment (iansu) [inline packages/scorenami-api/config/schema-map.js]: Is the value of `nfl_id` and `id` in `gameSummary` the same?

File: packages/scorenami-api/config/schema-map.js [hunk 20]
```diff
@@ -0,0 +1,152 @@
...
+    week: 'week',
+    year: 'year',
+    final: 'final',
+    home_score: 'homeScore',
+    away_score: 'awayScore'
+  },
+  teamGame: {
+    team: 'team',
+    opponent: 'opponent',
+    totfd: 'totalDrives',
```
Comment (iansu) [inline packages/scorenami-api/config/schema-map.js]: This should be `totalFirstDowns`.

File: packages/scorenami-api/config/schema-map.js [hunk 21]
```diff
@@ -0,0 +1,152 @@
...
+    totyds: 'totalYards',
+    pyds: 'passingYards',
+    ryds: 'runningYards',
```
Comment (iansu) [inline packages/scorenami-api/config/schema-map.js]: This should be `rushingYards`.

File: packages/scorenami-api/lib/pro-football-api.js [hunk 22]
```diff
@@ -0,0 +1,130 @@
...
+
+const translateGameSchema = dataAPI => {
+  const game = transformPropNames(dataAPI, 'game');
+  game.home = translateTeamGameDetailSchema(dataAPI.home);
+  game.away = translateTeamGameDetailSchema(dataAPI.away);
+
+  return game;
+};
+
+const translateGameSummariesSchema = dataAPI => {
```
Comment (iansu) [inline packages/scorenami-api/lib/pro-football-api.js]: This method should be named `translateGameSummarySchema`.

File: packages/scorenami-api/lib/pro-football-api.js [hunk 23]
```diff
@@ -0,0 +1,130 @@
...
+  return dataAPI.map(gameSummary => transformPropNames(gameSummary, 'gameSummary'));
+};
+
+const translateTeamGameDetailSchema = dataAPI => {
```
Comment (iansu) [inline packages/scorenami-api/lib/pro-football-api.js]: This method should be named `translateGameTeamDetailSchema` and the associated GraphQL type should be `GameTeamDetail`. This name change should also apply to any related methods, objects, etc.

File: packages/scorenami-api/lib/pro-football-api.js [hunk 24]
```diff
@@ -0,0 +1,130 @@
...
+  const stats = playIds.map(playId => {
+    const statData = transformPropNames(statsDataAPI[playId], type);
+
+    return Object.assign(statData, { playNumber: playId });
+  });
+
+  return stats;
+};
+
+const toSnakeCase = args => {
```
Comment (iansu) [inline packages/scorenami-api/lib/pro-football-api.js]: We're converting *from* snake case so this method should be `toCamelCase`.

File: packages/scorenami-api/schema/drive-schema.js [hunk 25]
```diff
@@ -0,0 +1,14 @@
+const schema = `
+  type Drive {
+    driveId: Int
+    quarter: Int
+    result: String
+    penaltyYards: Int
```
Comment (iansu) [inline packages/scorenami-api/schema/drive-schema.js]: I think all yardage values will have to be floats as you can have fractional yards. Double check the API to confirm that.

File: packages/scorenami-api/schema/game-schema.js [hunk 26]
```diff
@@ -13,6 +28,23 @@ const schema = `
     homeScore: Int
     awayScore: Int
   }
+
+  type TeamGame {
```
Comment (iansu) [inline packages/scorenami-api/schema/game-schema.js]: I think this type should be called `GameTeamSummary`.

File: packages/scorenami-api/config/schema-map.js (hunk 19 above)
Comment (Nosmohta) [reply packages/scorenami-api/config/schema-map.js]: The API uses id, nfl_id and nfl_game_id to reference the gameId.

File: packages/scorenami-api/lib/pro-football-api.js (hunk 23 above)
Comment (Nosmohta) [reply packages/scorenami-api/lib/pro-football-api.js]: When we integrate the `team` resolvers we will need four "game" types.  I have used `Game`, `GameSummary`, TeamGame` and `TeamGameSummary`.  I will change the function name to translateGameTeamSchema, unless you would prefer otherwise.

File: packages/scorenami-api/schema/drive-schema.js (hunk 25 above)
Comment (Nosmohta) [reply packages/scorenami-api/schema/drive-schema.js]: I double checked, and all data from the API for yard is an integer.

File: packages/scorenami-api/schema/game-schema.js (hunk 26 above)
Comment (Nosmohta) [reply packages/scorenami-api/schema/game-schema.js]: See comment above.  The team resolver returns similar data, but with no `drives` or `stats`. I have call that `TeamGameSummary` in the next branch.
